x/observer/keeper/grpc_query_prove.go

```diff
@@ -4,41 +4,63 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/btcsuite/btcutil"
 	"github.com/zeta-chain/zetacore/common"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	eth "github.com/ethereum/go-ethereum/common"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/zeta-chain/zetacore/x/observer/types"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+// Prove simply checks two things:
+// 1. the block header is available
+// 2. the proof is good
 func (k Keeper) Prove(c context.Context, req *types.QueryProveRequest) (*types.QueryProveResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 	ctx := sdk.UnwrapSDKContext(c)
 
-	blockHash := eth.HexToHash(req.BlockHash)
-	res, found := k.GetBlockHeader(ctx, blockHash.Bytes())
+	blockHash, err := common.StringToHash(req.ChainId, req.BlockHash)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	res, found := k.GetBlockHeader(ctx, blockHash)
 	if !found {
 		return nil, status.Error(codes.NotFound, "block header not found")
 	}
 
 	proven := false
 
-	val, err := req.Proof.Verify(res.Header, int(req.TxIndex))
+	txBytes, err := req.Proof.Verify(res.Header, int(req.TxIndex))
 	if err != nil && !common.IsErrorInvalidProof(err) {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 	if err == nil {
-		var txx ethtypes.Transaction
-		err = txx.UnmarshalBinary(val)
-		if err != nil {
-			return nil, status.Error(codes.Internal, fmt.Sprintf("failed to unmarshal transaction: %s", err))
+		if common.IsEVMChain(req.ChainId) {
```

Comment:
I think the current Ethereum proof system is for Ethereum specifically and not EVM, and we should use `IsEthereumChain` instead

Maybe @brewmaster012 can confirm

Same comment for the other places where we use `IsEVMChain`

Reply:
Good question. Current code base only accepts Ethereum block headers at [ValidateBasic](https://github.com/zeta-chain/node/blob/bitcoin-header-N-proof/x/observer/types/messages_add_block_header.go#L55)
We should explicitly restrict which chains are enabled in the source code in a consistent manner (on messages of `MsgAddBlockHeader`, `QueryProveRequest) and `MsgAddToOutTxTracker`)

Reply:
>We should explicitly restrict which chains are enabled in the source code in a consistent manner (on messages of MsgAddBlockHeader, QueryProveRequest) and MsgAddToOutTxTracker`)

I meant it would solve it if we replace `IsEVMChain` with `IsEthereumChain` in the different locations, isn't it?

Reply:
Current Ethereum proof is based and tested on Ethereum only, so most relaible on Ethereum. 
Most likely they also work for other EVM chains, if they are not messed up too much (deviating from go-ethereum too much). Klaytn might have problems, Polygon is unknown, BSC might be ok. 

For now I think IsEVMChain() because it basically means Ethereum/BSC.